### ISSUE TITLE
Is it possible to make custom loss function with more input than y_pred and y_true?

### ISSUE THREAD

**Issue by mr2dot4** [fchollet/keras#1074]
I would like to design a loss function base on the Ranking score, which is not directly derived from y_pred and y_true. 

My idea is to input two feature, that's say 64 + 64 = 128 concatenate together as the input layer and output the 1 unit regression of these two feature. 
I am using Muti-layer Perceptron structure.
Input: X and Xselect (64+64 unit)
Output: y (1 unit)
where Xselect is a feature selected from Xlibrary to compare with X. So I expect to get the output candidates by looping the X with the selections of X in library to make a rank. Then use those the rank list as the loss function by Mean Reciprocal Rank. 

In short, I need to calculate multiple forward propagation to create the rank list, and to make my loss function. I wonder if it is possible to do that in Keras? Or I should define the function in then?

**Comment by rpinsler** [comment]
I think I had a similar issue and solved it by currying, i.e. write a function that only takes the additional arguments as input (everything except y_pred and y_true) and returns another function that is compatible to the interface (i.e. only takes y_pred and y_true as input), see #1061.
However, it feels a little hacky, so I'd be happy to hear of a better solution.

**Comment by keunwoochoi** [comment]
Is there any update on it? I'm also interested in using ranking-based metric but not sure how could I put it into function.

**Comment by edmondja** [comment]
Any clean solution ? I am trying to implement MDN but this is a blocking issue for me.

**Comment by newbiesitl** [comment]
any update>

**Comment by fzyzcjy** [comment]
@newbiesitl Some links that may be helpful:

https://stackoverflow.com/questions/57897080/load-custom-loss-with-extra-input-in-keras
https://towardsdatascience.com/advanced-keras-constructing-complex-custom-losses-and-metrics-c07ca130a618
https://datascience.stackexchange.com/questions/25029/custom-loss-function-with-additional-parameter-in-keras
